testapi.pm

```diff
@@ -536,7 +541,14 @@ returns the base URL to contact the local os-autoinst service
 
 sub autoinst_url() {
     # move to backend?
-    return "http://10.0.2.2:" . (get_var("QEMUPORT") + 1);
+    if (get_var("CONNECT_PASSWORD")) {    #CONNECT_PASSWORD is used in HACLUSTER
+        use Net::Domain;
+        my $fqdn = Net::Domain::hostfqdn();
```

Comment:
It will not work if names resolution is broken (in HA tests I have a support server with DNS, so it always works).